grpc/core/src/main/java/io/helidon/grpc/core/model/BidiStreamingMethodHandlerSupplier.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.grpc.core.model;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.function.Supplier;
+
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * A supplier of {@link MethodHandler}s for bi-directional streaming gRPC methods.
+ *
+ * @author Jonathan Knight
+ */
+public class BidiStreamingMethodHandlerSupplier
```

Comment:
if we rename the annotation we should probably rename this class as well, although it is not as important as it's not used directly by the users.

Reply:
Which also raises the question of whether it should even be public (I believe Helidon preference is to make things package private unless they need to be publicly exposed)